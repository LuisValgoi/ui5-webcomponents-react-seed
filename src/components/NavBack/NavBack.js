import React from 'react';

import { Button, ButtonDesign, FlexBox } from '@ui5/webcomponents-react';
import { useHistory } from 'react-router-dom';
import { spacing } from '@ui5/webcomponents-react-base';

export default function NavBack({ text = 'Back' }) {
  const history = useHistory();

  return (
    <FlexBox style={spacing.sapUiTinyMargin}>
      <Button data-testid="navback-wrapper" design={ButtonDesign.Transparent} disabled={false} icon="nav-back" iconEnd={false} onClick={() => history.goBack()} submits={false}>
        {text}
      </Button>
    </FlexBox>
  );
}
