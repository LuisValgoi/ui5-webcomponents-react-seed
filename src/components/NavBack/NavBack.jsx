import { Button, FlexBox } from '@ui5/webcomponents-react';
import React from 'react';
import { useHistory } from 'react-router-dom';

export default function NavBack() {
  const history = useHistory();

  return (
    <FlexBox style={{ marginRight: '6px', marginTop: '6px', marginBottom: '6px', marginLeft: '6px' }}>
      <Button style={{ width: '100px' }} design="Default" disabled={false} icon="nav-back" iconEnd={false} onClick={() => history.goBack()} submits={false}>
        Back
      </Button>
    </FlexBox>
  );
}
