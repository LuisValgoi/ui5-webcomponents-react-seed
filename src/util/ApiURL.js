export default {
  GET_USER_LOGGED: '/v1/user/logged',
  GET_TODO_LIST: '/v1/todo/all',
  GET_TODO_BY_ID: '/v1/todo/:id'
};
