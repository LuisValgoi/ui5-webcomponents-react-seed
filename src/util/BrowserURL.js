export default {
  HOME: '/',
  TODO_LIST: '/todo/list',
  TODO_DETAIL: '/todo/:id/detail',
  NOT_FOUND: '/notFound',
  BUGGY: '/buggy',
  ANY: '/*'
};
