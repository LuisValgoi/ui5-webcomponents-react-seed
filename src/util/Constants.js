export default {
  REACT_QUERY: {
    CODES: {
      SUCCESS: 'success',
      LOADING: 'loading',
      ERROR: 'error',
    },
    KEYS: {
      RQ_GET_USER_LOGGED: 'RQ_GET_USER_LOGGED',
      RQ_GET_TODO_LIST: 'RQ_GET_TODO_LIST',
      GET_TODO_BY_ID: 'GET_TODO_BY_ID',
    },
  },
  SEED: {
    SELECTED_THEME: 'SELECTED_THEME',
  },
};
