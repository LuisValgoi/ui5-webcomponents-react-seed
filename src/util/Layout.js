import React from 'react';
import { Text } from '@ui5/webcomponents-react/lib/Text';

export const TextWrapper = ({ children, isOnHeader, style }) => {
  return (
    <Text
      tooltip={children}
      wrapping
      className={isOnHeader && 'text-header-color'}
      style={{ ...style, lineHeight: '20px' }}
    >
      {children}
    </Text>
  );
};
