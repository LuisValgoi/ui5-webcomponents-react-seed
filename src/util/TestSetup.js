import React from 'react';

import { render as rtlRender } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { rest } from 'msw';
import { setupServer } from 'msw/node';

import BrowserProvider from './browser/BrowserProvider';

const render = (ui, { route = BrowserProvider.getUrl('HOME'), ...renderOptions } = {}) => {
  const WrapperProvider = ({ ...props }) => {
    const history = createMemoryHistory({ initialEntries: [route] });
    return (
      <BrowserRouter>
        <Router history={history}>{props.children}</Router>
      </BrowserRouter>
    );
  };

  return rtlRender(ui, { wrapper: WrapperProvider, ...renderOptions });
};

const serverCustom = (apiUrl, data) => {
  return setupServer(
    rest.get(`*${apiUrl}`, (req, res, ctx) => {
      return res(ctx.json(data));
    }),
  );
};

export * from '@testing-library/react';

export { render, serverCustom };
