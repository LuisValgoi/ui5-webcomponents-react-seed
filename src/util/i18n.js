import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

import LanguageDetector from 'i18next-browser-languagedetector';

import translationEN from '../locales/en/translation.json';
import translationPT from '../locales/pt/translation.json';

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    debug: true,
    keySeparator: false,
    interpolation: {
      escapeValue: false,
    },
    resources: {
      en: {
        translation: translationEN,
      },
      pt: {
        translation: translationPT,
      },
    },
    react: {
      useSuspense: false,
    },
  });

export default i18n;
