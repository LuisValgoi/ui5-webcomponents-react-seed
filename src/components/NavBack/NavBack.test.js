import React from 'react';

import '@testing-library/jest-dom/extend-expect';
import { fireEvent, render, screen } from '../../util/TestSetup';

import NavBack from './NavBack';

describe('NavBack.js Test Suite', () => {
  beforeEach(() => {
    render(<NavBack />, { route: 'TODO_EDIT' });
  });

  test('should contains Back text if nothing is passed', () => {
    const navback = screen.getByTestId('navback-wrapper');

    expect(navback).toHaveTextContent('Back');
  });

  test('should contains transparent button', () => {
    const navback = screen.getByTestId('navback-wrapper');

    expect(navback).toHaveProperty('design', 'Transparent');
  });

  test('should redirect to home if pressed', async () => {
    const navback = screen.getByTestId('navback-wrapper');

    await fireEvent.click(navback);

    const currentURL = window.location.pathname;

    expect(currentURL).toEqual('/');
  });

  test('should contains Nav Back text if is passed', () => {
    render(<NavBack text="Nav Back" />);

    const navback = screen.getAllByTestId('navback-wrapper')[1];

    expect(navback).toHaveTextContent('Nav Back');
  });
});
