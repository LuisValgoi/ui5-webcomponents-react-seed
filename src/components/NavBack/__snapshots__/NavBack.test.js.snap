// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBack.js Test Suite should match snapshot 1`] = `
<ui5-button
  data-testid="navback-wrapper"
  design="Transparent"
  icon="nav-back"
>
  Back
</ui5-button>
`;
